### PR TITLE
RA-501

### PR DESCRIPTION
**Pull request by aoller**

### CODE REVIEW

File: VAS.Tests/Core/TestExtensionMethods.cs [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+		{
+			CustomDummyClassForTest [] array = new CustomDummyClassForTest [] {
+				new CustomDummyClassForTest { Index = 10 },
+				new CustomDummyClassForTest { Index = 6 },
+				new CustomDummyClassForTest { Index = 4 },
+				new CustomDummyClassForTest { Index = 5 }
+ 			};
+
+			CustomDummyClassForTest [] arrayDesc = array.Sort (c => c.Index, true).ToArray ();
+			CustomDummyClassForTest [] arrayAsc = array.Sort (c => c.Index, false).ToArray ();
```
Comment (aameijide) [inline VAS.Tests/Core/TestExtensionMethods.cs]: Split in 2 test, one for ascending and other for descending

File: VAS.Tests/Core/TestExtensionMethods.cs [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+			DateTime dt4 = dt3.AddSeconds (1);
+			StorableBase [] array = new StorableBase [] {
+				new StorableBase { CreationDate = dt2, ID = Guid.NewGuid()},
+				new StorableBase { CreationDate = dt4, ID = Guid.NewGuid() },
+				new StorableBase { CreationDate = dt1, ID = Guid.NewGuid() },
+				new StorableBase { CreationDate = dt3, ID = Guid.NewGuid() }
+ 			};
+
+			StorableBase [] arrayDesc = array.SortByCreationDate (true).ToArray ();
+			StorableBase [] arrayAsc = array.SortByCreationDate (false).ToArray ();
```
Comment (aameijide) [inline VAS.Tests/Core/TestExtensionMethods.cs]: Split in 2 test, one for ascending and other for descending

File: VAS.Tests/Core/TestExtensionMethods.cs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+			CustomDummyClassForTest [] arrayAsc = array.Sort (c => c.Index, false).ToArray ();
+
+			Assert.AreEqual (10, arrayDesc [0].Index);
+			Assert.AreEqual (6, arrayDesc [1].Index);
+			Assert.AreEqual (5, arrayDesc [2].Index);
+			Assert.AreEqual (4, arrayDesc [3].Index);
+			Assert.AreEqual (4, arrayAsc [0].Index);
+			Assert.AreEqual (5, arrayAsc [1].Index);
+			Assert.AreEqual (6, arrayAsc [2].Index);
+			Assert.AreEqual (10, arrayAsc [3].Index);
```
Comment (aameijide) [inline VAS.Tests/Core/TestExtensionMethods.cs]: You can use CollectionAssert.AreEqual, that let you to check collection with the same order.

File: VAS.Tests/Core/TestExtensionMethods.cs [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+			StorableBase [] arrayAsc = array.SortByCreationDate (false).ToArray ();
+
+			Assert.AreEqual (array [1].ID, arrayDesc [0].ID);
+			Assert.AreEqual (array [3].ID, arrayDesc [1].ID);
+			Assert.AreEqual (array [0].ID, arrayDesc [2].ID);
+			Assert.AreEqual (array [2].ID, arrayDesc [3].ID);
+			Assert.AreEqual (array [2].ID, arrayAsc [0].ID);
+			Assert.AreEqual (array [0].ID, arrayAsc [1].ID);
+			Assert.AreEqual (array [3].ID, arrayAsc [2].ID);
+			Assert.AreEqual (array [1].ID, arrayAsc [3].ID);
```
Comment (aameijide) [inline VAS.Tests/Core/TestExtensionMethods.cs]: You can use CollectionAssert

File: VAS.Tests/Core/TestExtensionMethods.cs [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+		}
+	}
+
+	class CustomDummyClassForTest
```
Comment (aameijide) [inline VAS.Tests/Core/TestExtensionMethods.cs]: Move to VAS.Test.Utils, there are all dummy classes for test

File: VAS.Tests/Utils.cs [hunk 6]
```diff
@@ -409,5 +410,64 @@ public static string GetEmbeddedResourceTextFile (string resourceId)
 			return result;
 		}
 	}
+
+	class CustomDummyClassForTest
+	{
+		public int Index { get; set; }
+	}
+
+	class StorableBaseComparer : IComparer<StorableBase>, IComparer
```
Comment (aameijide) [inline VAS.Tests/Utils.cs]: Can we mix both comparer implementing only `IComparer` and in Compare method with object parameters call to the method with correspondent parameters?

Comment (aoller) [inline VAS.Tests/Utils.cs]: As we discussed, if we need to sort same object by a different logic, with your proposal the implementation isn't so clear.

**Pull request merged**